TreeMaker/python/makeTreeFromMiniAOD_cff.py

```diff
@@ -203,6 +203,15 @@ def makeTreeFromMiniAOD(self,process):
         self.VectorRecoCand.append("genTops(GenTops)")
         self.VarsDouble.append("genTops:weight(GenTopWeight)")
 
+        #Event weight to fix FastSim pre-UL bug
+        if self.fastsim:
+            process.FastSimWeightPR31285To36122 = cms.EDProducer("FastSimWeightPR31285To36122",
+                genCollection = cms.InputTag("prunedGenParticles"),
+                genJetTag = cms.InputTag('slimmedGenJets'),
+                recJetTag = cms.InputTag("slimmedJets")
+            )
+            self.VarsDouble.append("FastSimWeightPR31285To36122:EventWeight(FastSimWeightPR31285To36122)")
```

Comment:
the branch name could be simplified to just `FastSimWeight`

Reply:
Yes, although the weight is only appropriate for productions created with releases created in the specified range of PR's. The weight *can* be used without creating a bias for other FastSim samples, but it will hurt the statistics unnecessarily.